File: app/src/main/java/com/anokmik/tripassistant/databinding/components/TextInputLayoutComponent.java
package com.anokmik.tripassistant.databinding.components;

import android.databinding.BindingAdapter;
import android.graphics.drawable.Drawable;
import android.support.design.widget.TextInputLayout;
import android.text.Editable;
import android.text.TextUtils;
import android.widget.EditText;

public final class TextInputLayoutComponent {

    @BindingAdapter({"error", "showError"})
    public void setError(TextInputLayout view, String error, boolean showError) {
        view.setError(showError ? error : null);
    }

    @BindingAdapter("editable")
    public void setEditable(TextInputLayout view, boolean isEditable) {
        view.setFocusable(isEditable);
        EditText editText = view.getEditText();
        if (editText != null) {
            editText.setCursorVisible(isEditable);
            editText.setFocusable(isEditable);
            editText.setFocusableInTouchMode(isEditable);
            Drawable background = editText.getBackground();
            if (background != null) {
                background.setAlpha(isEditable ? 255 : 0);
            }
            if (isEditable) {
                Editable editableText = editText.getText();
                int selectionStart = editText.getSelectionStart();
                int selectionEnd = editText.getSelectionEnd();
                if (!TextUtils.isEmpty(editableText) && (selectionStart == 0) && (selectionEnd == 0)) {
                    editText.setSelection(editableText.length());
                }
            }
        }
    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/databinding/components/TextInputLayoutComponent.kt
package com.anokmik.tripassistant.databinding.components

import android.databinding.BindingAdapter
import android.support.design.widget.TextInputLayout
import android.text.TextUtils

class TextInputLayoutComponent {

    @BindingAdapter("error", "showError")
    fun setError(view: TextInputLayout, error: String, showError: Boolean) {
        view.error = if (showError) error else null
    }

    @BindingAdapter("editable")
    fun setEditable(view: TextInputLayout, isEditable: Boolean) {
        view.isFocusable = isEditable
        view.editText?.apply {
            isCursorVisible = isEditable
            isFocusable = isEditable
            isFocusableInTouchMode = isEditable
            background?.apply {
                alpha = if (isEditable) 255 else 0
            }
            if (isEditable) {
                val editableText = text
                if (!TextUtils.isEmpty(editableText) && (selectionStart == 0) && (selectionEnd == 0)) {
                    setSelection(editableText.length)
                }
            }
        }
    }

}